instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/HttpUrlReplacements.java

```diff
@@ -76,7 +76,6 @@ public static int replacementForContentLength(URLConnection connection) {
         return replace(connection, () -> connection.getContentLength());
     }
 
-    @SuppressLint("NewApi")
```

Comment:
This is a simple example that would require:

```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
	return replace(connection, () -> connection.getContentLengthLong());
} else {
	return -1
}
```

And the linter wouldn't need to be suppressed but its out of the scope of this PR, just sharing.

Reply:
That's right, I wasn't hoping for animal sniffer to be a replacement for these kinds of approaches which we should still use IMO. We do so already in some services like [this one](https://github.com/open-telemetry/opentelemetry-android/blob/c10adbaf11dbc415190d5fdf8dcf5183485d3ef6/core/src/main/java/io/opentelemetry/android/internal/services/CacheStorage.java#L47) for example.

The example highlighted here is a special one because the class `HttpUrlReplacements` is meant to be used during bytecode instrumentation to replace existing calls to the original methods, so for this case, the method `HttpUrlReplacements.replacementForContentLengthLong` will only be used in apps that are already calling the original method [getContentLengthLong()](https://developer.android.com/reference/java/net/URLConnection#getContentLengthLong()), so it implicitly means that their minSdk is at least 24, otherwise they'd get an AGP compilation error.

That being said, I'm not fully sure what would happen in this case when animal sniffer is involved, I'm assuming it will complain because the signature we're going to use is of API 21 and also because I don't think this method has nothing to do with the desugaring corelibs either, so what I can propose is to follow what @surbhiia mentioned earlier about waiting for the animal sniffer integration before removing the "NewApi" checks just to make sure it behaves the same way and wouldn't just ignore these types of scenarios which are a potential runtime crash in most cases.

Reply:
makes sense.